app/services/manifest_submitter.rb

```diff
@@ -0,0 +1,26 @@
+class ManifestSubmitter
+  include Pundit
+  attr_reader :args
+
+  def initialize(args)
+    @args = args
+  end
+
+  def perform
+    output_stream = StreamLogger.new(Rails.logger)
+
+    manifest = Manifest.find_by_uuid_or_tracking_number!(args[:manifest_id])
+    authorize manifest, :can_submit?
```

Comment:
I have great difficulty taking a controller and making it load data and apply some business logic to it. The responsibilities get muddled and my controller must understand all these concepts to do its multi-pronged job.

Then again, I like to reuse services, not controllers, so I give the services all the responsibility to maintain consistent business rules and leave the controllers with the sole responsibility of translating between the outside world and something valid in our business model.

So it comes down to what's acceptable? Do we make the controller do lots of things and let the service assume all of the data coming to it has been checked, or do we let the controller do its one thing and refactor whatever service it calls as needed to keep the responsibilities clear? I'm going to push hard against having big-ol'-beefy controllers (where's the Trogdor emoji?) even if all of the Pundit docs say "throw everything you need for business rule authorization into your controller!"